pkg/uid/from_etc.go

```diff
@@ -0,0 +1,60 @@
+// Copyright 2016 The rkt Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// For how the uidshift and uidcount are generated please check:
+// http://cgit.freedesktop.org/systemd/systemd/commit/?id=03cfe0d51499e86b1573d1
+
+package uid
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/coreos/rkt/pkg/group"
+	"github.com/coreos/rkt/pkg/passwd"
+	"github.com/hashicorp/errwrap"
+)
+
+// FromEtc is the struct that generates a uid/gid by parsing etc/passwd,
+// and etc/group relative from the given Root path looking for the given Name.
+type FromEtc struct {
```

Comment:
@jonboulle I now have a better idea of your position in "_test". Here is how everything should be tested without exporting everything:
```go
package uid

type Generator interface{
  ....
}

type fromEtc struct {...}

var _ Generator = &fromEtc{}

func FromEtc(....) Generator {
  ...
  return &fromEtc{...}
}
```
and the tests should be like:
```go
package uid_test

func TestCreatesIDGeneratorFromEtc(t *testing.T) {
  g := uid.FromEtc(...)

  if g.UID != expected {
  ...
  }
  
  if g.GID != expected {
  ...
  }
```
and this function should be repeated for all exported constructors or a table-driven test should be used.

This way you test everything and you only export the interface and constructors.